### PR TITLE
Overhaul root system type detection

### PR DESCRIPTION
**Pull request by lgoettgens**
The function `cartan_type_with_ordering` errored and gave wrong results on some inputs. This PR completely rewrites the function and adds exhaustive tests.
Furthermore, all of the constructors of root system related types now take the same types of arguments.
And there are now more ways to print a Dynkin diagram. These were added here to be able to provide some visual tests for `cartan_type_with_ordering`.

### CODE REVIEW

File: experimental/LieAlgebras/test/WeylGroup-test.jl [hunk 1]
```diff
@@ -55,6 +55,7 @@ include(
     ("B4", weyl_group(root_system(:B, 4))),
     ("D5", weyl_group(cartan_matrix(:D, 5))),
     ("F4+G2", weyl_group((:F, 4), (:G, 2))),
+    ("E6+C3", weyl_group([(:E, 6), (:C, 2)])),
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/test/WeylGroup-test.jl]:
```suggestion
    ("E6+C3", weyl_group([(:E, 6), (:C, 3)])),
```

File: experimental/LieAlgebras/src/RootSystem.jl [hunk 2]
```diff
@@ -280,10 +284,29 @@ function rank(R::RootSystem)
 end
 
 function root_system_type(R::RootSystem)
-  @req has_root_system_type(R) "root system type not defined"
```
Comment (felix-roehrich) [inline experimental/LieAlgebras/src/RootSystem.jl:283]: What happens when the root system type is not known ?

Comment (lgoettgens) [reply experimental/LieAlgebras/src/RootSystem.jl:283]: The next line errors

File: experimental/LieAlgebras/test/CartanMatrix-test.jl
Comment (felix-roehrich) [inline experimental/LieAlgebras/test/CartanMatrix-test.jl:1]: For the tests there is a lot of repetition. Maybe put the common tests into a test function and test special cases like B2/C2 extra. I don't know how I feel about the tests for D4 and E6: The returned ordering could be said to be an implementation detail, in the sense that if ordering is changed by changes to the algorithm this not a breaking change; in this case that should be clearly documented.

File: experimental/LieAlgebras/src/RootSystem.jl (hunk 2 above)
Comment (felix-roehrich) [reply experimental/LieAlgebras/src/RootSystem.jl:283]: Maybe a more explicit error would be good for the user to know why this fails. Getting a "not defined" (?) error is not really helpful.

**Comment by lgoettgens** [comment]
Everything should be addressed now

**Review by felix-roehrich — APPROVED** [review]
Thanks. Lgtm.

**Pull request merged**